keras/backend/tensorflow_backend.py

```diff
@@ -870,8 +874,7 @@ def batch_set_value(tuples):
             `value` should be a Numpy array.
     '''
     if tuples:
-        ops = [tf.assign(x, np.asarray(value)) for x, value in tuples]
-        get_session().run(ops)
+        [set_value(x,np.asarray(value)) for x, value in tuples]
```

Comment:
I didn't catch this at first, but this is a problem. You are running `sess.run()` N times whereas you should only run it one time (that's the entire point of `batch_set_value`. Running it N times is extremely inefficient when you have many weights (large models can take minutes to load instead of ~1 sec).